rust/crd/src/authentication.rs

```diff
@@ -0,0 +1,290 @@
+use serde::{Deserialize, Serialize};
+use stackable_operator::builder::{
+    SecretOperatorVolumeSourceBuilder, VolumeBuilder, VolumeMountBuilder,
+};
+use stackable_operator::k8s_openapi::api::core::v1::{CSIVolumeSource, Volume, VolumeMount};
+
+use stackable_operator::kube::CustomResource;
+use stackable_operator::schemars::{self, JsonSchema};
+
+#[derive(Clone, CustomResource, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[kube(
+    group = "authentication.stackable.tech",
+    version = "v1alpha1",
+    kind = "AuthenticationClass",
+    plural = "authenticationclasses",
+    crates(
+        kube_core = "stackable_operator::kube::core",
+        k8s_openapi = "stackable_operator::k8s_openapi",
+        schemars = "stackable_operator::schemars"
+    )
+)]
+#[serde(rename_all = "camelCase")]
+pub struct AuthenticationClassSpec {
+    /// Protocol used for authentication
+    pub protocol: AuthenticationClassProtocol,
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum AuthenticationClassProtocol {
+    Ldap(AuthenticationClassLdap),
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AuthenticationClassLdap {
+    /// Hostname of the LDAP server
+    pub hostname: String,
+    /// Port of the LDAP server
+    #[serde(default = "default_port")]
+    pub port: u16,
+    /// LDAP search base
+    #[serde(default)]
+    pub search_base: String,
+    /// LDAP query to filter users
+    #[serde(default)]
+    pub search_filter: String,
+    /// The name of the username field
+    #[serde(default = "default_uid_field")]
+    pub uid_field: String,
```

Comment:
All of these field are actually used by the superset operator at the moment. The other option would be to hardcode them in the superset operator but that may would imply that we e.g. don't support non-default openldap installations or Microsoft AD.

But i made all non-essential attributes optional (by providing a default for openldap) so that the normal user does not need to know about them